### PR TITLE
[Win10] Implementation UWP own power management and storage provider

### PR DESCRIPTION
**Pull request by afedchin**
let's make things right :)

### CODE REVIEW

File: xbmc/storage/windows/Win32StorageProvider.cpp [hunk 1]
```diff
@@ -56,46 +53,119 @@ void CWin32StorageProvider::Initialize()
...
 }
 
 std::vector<std::string > CWin32StorageProvider::GetDiskUsage()
 {
-  return CWIN32Util::GetDiskUsage();
+  std::vector<std::string> result;
+  ULARGE_INTEGER ULTotal = { { 0 } };
+  ULARGE_INTEGER ULTotalFree = { { 0 } };
+
+  std::unique_ptr<wchar_t> pcBuffer;
```
Comment (Paxxi) [inline xbmc/storage/windows/Win32StorageProvider.cpp]: `std::unique_ptr<wchar_t[]>` or it'll use the wrong deleter

File: xbmc/storage/windows/Win32StorageProvider.cpp [hunk 2]
```diff
@@ -105,6 +175,188 @@ bool CWin32StorageProvider::PumpDriveChangeEvents(IStorageEventsCallback *callba
   return b;
 }
 
+void CWin32StorageProvider::GetDrivesByType(VECSOURCES &localDrives, Drive_Types eDriveType, bool bonlywithmedia)
+{
+  using KODI::PLATFORM::WINDOWS::FromW;
+
+  WCHAR* pcBuffer= NULL;
```
Comment (Paxxi) [inline xbmc/storage/windows/Win32StorageProvider.cpp]: might want to use unique_ptr here as well but it's fine to keep as is as well.

File: xbmc/powermanagement/PowerManager.cpp [hunk 3]
```diff
@@ -52,9 +52,11 @@
 #include "linux/LogindUPowerSyscall.h"
 #include "linux/UPowerSyscall.h"
 #endif // HAS_DBUS
-#elif defined(TARGET_WINDOWS)
+#elif defined(TARGET_WINDOWS_DESKTOP)
 #include "powermanagement/windows/Win32PowerSyscall.h"
 extern HWND g_hWnd;
+#elif defined(TARGET_WINDOWS_STORE)
+#include "powermanagement/win10/PowerSyscallWin10.h"
```
Comment (Paxxi) [inline xbmc/powermanagement/PowerManager.cpp]: maybe name the file `Win10PowerSyscall.h` to keep it consistent, or rename the Win32 file, both works

File: xbmc/powermanagement/win10/PowerSyscallWin10.cpp [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "PowerSyscallWin10.h"
+
+using namespace Windows::Devices::Power;
+
+CPowerSyscall::CPowerSyscall()
```
Comment (Paxxi) [inline xbmc/powermanagement/win10/PowerSyscallWin10.cpp]: `CPowerSyscall::CPowerSyscall() = default;`

File: xbmc/powermanagement/win10/PowerSyscallWin10.cpp [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+{
+}
+
+CPowerSyscall::~CPowerSyscall()
```
Comment (Paxxi) [inline xbmc/powermanagement/win10/PowerSyscallWin10.cpp]: Same

File: xbmc/powermanagement/windows/Win32PowerSyscall.cpp [hunk 6]
```diff
@@ -100,7 +157,11 @@ bool CWin32PowerSyscall::CanReboot()
 
 int CWin32PowerSyscall::BatteryLevel()
 {
-  return CWIN32Util::BatteryLevel();
+  int result = 0;
```
Comment (Paxxi) [inline xbmc/powermanagement/windows/Win32PowerSyscall.cpp]: might as well just `return 0` as this isn't assigned anywhere else and it's such a small function

**Review by Paxxi — COMMENTED** [review]
Looks good to in general.

Fix the `unique_ptr<wchar_t[]>` and filename and it's good to go.

File: xbmc/storage/windows/Win32StorageProvider.cpp (hunk 1 above)
Comment (afedchin) [reply xbmc/storage/windows/Win32StorageProvider.cpp]: i'll fix this but it was in original code see https://github.com/xbmc/xbmc/pull/13254/files#diff-504bcfd2abeae7af84f0ab527d81e08fL396

**Comment by afedchin** [comment]
jenkins is happy see [BuildMulti-PR-Manually#123](http://jenkins.kodi.tv/view/All/job/BuildMulti-PR-Manually/123/)

**Pull request merged**